### PR TITLE
Fixes html_url property in filing end points

### LINKED ISSUE #2533: Add html_url value for /efile/filings/ and fix missing html_url for /filings
We've received a few comments recently about people looking for the links to the web-based view of filings. While we have a place to include these in the main filings tables, the `html_url` is currently `null` for many filings, though not all. (e.g. https://api.open.fec.gov/v1/efile/filings/?api_key=DEMO_KEY&form_type=F3). Also, it's completely absent from the `/efile/filings/` endpoint. 

(Note: It seems to work fine on the `/reports/` endpoints, which is why it shows up in the details panels. Those are populated by additional calls to that endpoint, whereas the table itself is just the filings endpoint).

**Completion criteria:**
- [ ] Fix missing `html_url` values on `/filings/`
- [ ] Add `html_url` property on `/efile/filings/`

### PR DESCRIPTION
**Pull request by rjayasekera**
Fixes #2533 

This change set fixes the html_url property in the filings end point and adds the property to the efile/filings end point.

**NOTE**

This requires running `update_schemas` in the database.

### CODE REVIEW

**Comment by ccostino** [comment]
@jontours or @LindsayYoung:  We exposed the `html_url` property in the `EFilingsSchema` but we weren't sure if it need to be exposed and added to the `BaseEfileSchema` - do either of you have a sense of whether or not that is needed?

**Comment by noahmanger** [comment]
Just checking: what will the behavior be for paper filings? They'll still have `html_url: null` yes?

**Comment by jontours** [comment]
In these instances of unwinding the Gordian Knot that is schemas.py it's best to map the right Schema object from the endpoint view.

**Comment by jontours** [comment]
@noahmanger for paper it depends, if the file number is less than 1, then yes it will be null.

File: data/sql_updates/filings.sql [hunk 1]
```diff
@@ -74,10 +74,7 @@ with filings as (
             filing_history.form_tp
         ) as pdf_url,
         means_filed(begin_image_num) as means_filed,
-        case when means_filed(begin_image_num) = 'e-file' then
-          report_html_url(means_filed(begin_image_num), filing_history.cand_cmte_id::text, filing_history.file_num::text)
-          else null::text
-        end as html_url,
+        report_html_url(means_filed(begin_image_num), filing_history.cand_cmte_id::text, filing_history.file_num::text) as html_url,
```
Comment (jontours) [inline data/sql_updates/filings.sql]: Thanks for cleaning up after my forgetfulness.

File: webservices/schemas.py [hunk 2]
```diff
@@ -699,6 +699,7 @@ def remove_fec_url(self, obj):
     fields={
         'beginning_image_number': ma.fields.Str(),
         'ending_image_number': ma.fields.Str(),
+        'html_url': ma.fields.Str(),
```
Comment (jontours) [inline webservices/schemas.py]: There is a question about if this is the right schema, it is.   The other one that is similarly named are schemas that are used to build the efile endpoints, and shouldn't be referenced directly.  Schemas.py is a bit of a mess.

**Comment by PaulClark2** [comment]
A follow-up on @noahmanger's questions and @jontours's response. In the data, the file numbers of paper filings can be negative. This means we do not yet have a .fec file for that report. In cases when the file number is negative is the return from the function still `html_url: null`?

**Comment by jontours** [comment]
Yes @PaulClark2.

File: webservices/schemas.py (hunk 2 above)
Comment (ccostino) [inline webservices/schemas.py]: Okay excellent, thanks!  So we shouldn't worry about the other schema at all then?

**Pull request merged**